<h2 style="color:red">Descriptive Statistics</h2>

In [1]:
import pandas as pd

In [5]:
data = {'Name' : ["Sksham","Ram","Raghav","Priya","Kumar","Rajat","Neha","Sushant","Sunil","Santosh"],
       'Salary' : [30000,34000,30200,23400,32211,34532,66341,42321,24354,34213],
       'Country' : ["India","Canada","USA","Nepal","India","USA","Dubai","India","Canada","Italy"]}

In [10]:
data_df  = pd.DataFrame(data)

In [11]:
data_df

Name  Salary Country
0   Sksham   30000   India
1      Ram   34000  Canada
2   Raghav   30200     USA
3    Priya   23400   Nepal
4    Kumar   32211   India
5    Rajat   34532     USA
6     Neha   66341   Dubai
7  Sushant   42321   India
8    Sunil   24354  Canada
9  Santosh   34213   Italy

In [12]:
data_df.to_csv('Employee_data.csv')

In [17]:
df = pd.read_csv('Employee_data.csv')

In [18]:
df

Unnamed: 0     Name  Salary Country
0           0   Sksham   30000   India
1           1      Ram   34000  Canada
2           2   Raghav   30200     USA
3           3    Priya   23400   Nepal
4           4    Kumar   32211   India
5           5    Rajat   34532     USA
6           6     Neha   66341   Dubai
7           7  Sushant   42321   India
8           8    Sunil   24354  Canada
9           9  Santosh   34213   Italy

In [21]:
emp = df.drop(['Unnamed: 0'], axis=1)

In [22]:
emp

Name  Salary Country
0   Sksham   30000   India
1      Ram   34000  Canada
2   Raghav   30200     USA
3    Priya   23400   Nepal
4    Kumar   32211   India
5    Rajat   34532     USA
6     Neha   66341   Dubai
7  Sushant   42321   India
8    Sunil   24354  Canada
9  Santosh   34213   Italy

emp.info()

In [24]:
emp.describe()

Salary
count     10.000000
mean   35157.200000
std    12207.294876
min    23400.000000
25%    30050.000000
50%    33105.500000
75%    34452.250000
max    66341.000000

<h2 style="color:red">Measure of Central Tendancy</h2>

In [32]:
emp['Salary'].mean()

35157.2

In [33]:
emp['Salary'].median()

33105.5

In [34]:
emp['Salary'].mode()

0    23400
1    24354
2    30000
3    30200
4    32211
5    34000
6    34213
7    34532
8    42321
9    66341
dtype: int64

In [36]:
emp['Salary'].sum()

351572

In [37]:
emp['Salary'].min()

23400

In [38]:
emp['Salary'].max()

66341

In [39]:
emp['Salary'].count()

10

In [47]:
emp.groupby(['Country'])['Salary'].sum()

Country
Canada     58354
Dubai      66341
India     104532
Italy      34213
Nepal      23400
USA        64732
Name: Salary, dtype: int64

In [48]:
emp.groupby(['Country']).count()

Name  Salary
Country              
Canada      2       2
Dubai       1       1
India       3       3
Italy       1       1
Nepal       1       1
USA         2       2

<h2 style="color:red">Measure of Variability</h2>

In [49]:
# standard deviation

emp['Salary'].std()

12207.294875515123

In [50]:
# variance of Salary

emp['Salary'].var()

149018048.1777778

<h2 style="color:red">Measure of Symmetry</h2>

In [52]:
emp_skew = emp.skew(axis = 0, skipna = True)
emp_skew

Salary    2.101034
dtype: float64

In [57]:
# the skewness is positive so 

In [53]:
birth_rate = pd.read_csv("BirthWeight.csv")

In [54]:
birth_rate

Infant ID  Gestational Age (Weeks)  Birth Weight (Grams)
0           1                     34.7                  1895
1           2                     36.0                  2030
2           3                     29.3                  1440
3           4                     40.1                  2835
4           5                     35.7                  3090
5           6                     42.4                  3827
6           7                     40.3                  3260
7           8                     37.3                  2690
8           9                     40.9                  3285
9          10                     38.3                  2920
10         11                     38.5                  3430
11         12                     41.4                  3657
12         13                     39.7                  3685
13         14                     39.7                  3345
14         15                     41.1                  3260
15         16                     38.0                  2680
16         17                     38.7                  2005

In [55]:
birth_rate.set_index('Infant ID', inplace = True)

In [56]:
birth_rate.head()

Gestational Age (Weeks)  Birth Weight (Grams)
Infant ID                                               
1                             34.7                  1895
2                             36.0                  2030
3                             29.3                  1440
4                             40.1                  2835
5                             35.7                  3090

In [59]:
birth_rate.cov()

# to check the covariance 


Gestational Age (Weeks)  Birth Weight (Grams)
Gestational Age (Weeks)                 9.963824              1798.025
Birth Weight (Grams)                 1798.025000            485478.750

In [61]:
birth_rate.corr(method = "pearson")


Gestational Age (Weeks)  Birth Weight (Grams)
Gestational Age (Weeks)                 1.000000              0.817519
Birth Weight (Grams)                    0.817519              1.000000

**Covariance indicates that there is correlation exists between two.**

**correlaton coefficient of Gestational Age (Weeks) and Birth Weight (Grams) is (0.817519 ) positive and strong.**

In [62]:
# importing the required libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from scipy.stats import skew
from scipy.stats import kurtosis

In [63]:
# to display all columns
pd.set_option("display.max_columns", None)

# to display numbers in float value upto two decimal points
pd.options.display.float_format = "{:,.2f}".format

In [64]:
diamond = pd.read_csv("diamonds.csv")

In [65]:
diamond

id  carat        cut color clarity  depth  table  price    x    y  \
0          1   0.23      Ideal     E     SI2  61.50  55.00    326 3.95 3.98   
1          2   0.21    Premium     E     SI1  59.80  61.00    326 3.89 3.84   
2          3   0.23       Good     E     VS1  56.90  65.00    327 4.05 4.07   
3          4   0.29    Premium     I     VS2  62.40  58.00    334 4.20 4.23   
4          5   0.31       Good     J     SI2  63.30  58.00    335 4.34 4.35   
...      ...    ...        ...   ...     ...    ...    ...    ...  ...  ...   
53935  53936   0.72      Ideal     D     SI1  60.80  57.00   2757 5.75 5.76   
53936  53937   0.72       Good     D     SI1  63.10  55.00   2757 5.69 5.75   
53937  53938   0.70  Very Good     D     SI1  62.80  60.00   2757 5.66 5.68   
53938  53939   0.86    Premium     H     SI2  61.00  58.00   2757 6.15 6.12   
53939  53940   0.75      Ideal     D     SI2  62.20  55.00   2757 5.83 5.87   

         z  
0     2.43  
1     2.31  
2     2.31  
3     2.63  
4     2.75  
...    ...  
53935 3.50  
53936 3.61  
53937 3.56  
53938 3.74  
53939 3.64  

[53940 rows x 11 columns]

In [66]:
diamond.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53940 entries, 0 to 53939
Data columns (total 11 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   id       53940 non-null  int64  
 1   carat    53940 non-null  float64
 2   cut      53940 non-null  object 
 3   color    53940 non-null  object 
 4   clarity  53940 non-null  object 
 5   depth    53940 non-null  float64
 6   table    53940 non-null  float64
 7   price    53940 non-null  int64  
 8   x        53940 non-null  float64
 9   y        53940 non-null  float64
 10  z        53940 non-null  float64
dtypes: float64(6), int64(2), object(3)
memory usage: 4.5+ MB


In [68]:
# dont need ID column
dim_df = diamond.drop(['id'], axis = 1)

# drop all alpha-numeric columns

for col in dim_df:
    if dim_df[col].dtype == 'object':
        dim_df = dim_df.drop([col], axis = 1)
        
new_df = dim_df.describe()

# to change name 50% to median/50% in describe table
new_df = new_df.rename(index={'50%' : 'median/50%'})
new_df


# 25% = in my dataset 25% values are less than 0.40

carat     depth     table     price         x         y  \
count      53,940.00 53,940.00 53,940.00 53,940.00 53,940.00 53,940.00   
mean            0.80     61.75     57.46  3,932.80      5.73      5.73   
std             0.47      1.43      2.23  3,989.44      1.12      1.14   
min             0.20     43.00     43.00    326.00      0.00      0.00   
25%             0.40     61.00     56.00    950.00      4.71      4.72   
median/50%      0.70     61.80     57.00  2,401.00      5.70      5.71   
75%             1.04     62.50     59.00  5,324.25      6.54      6.54   
max             5.01     79.00     95.00 18,823.00     10.74     58.90   

                   z  
count      53,940.00  
mean            3.54  
std             0.71  
min             0.00  
25%             2.91  
median/50%      3.53  
75%             4.04  
max            31.80

**I want add variance in above table**

In [73]:
# NOT IN PRACTICAL




# calculating variance seperately
var_df = dim_df.var()

# creating empty list
varlist = []

for col in dim_df.columns:
    if dim_df[col].dtype == 'object':
        continue
    varlist.append(round(dim_df[col],5))
    
df = pd.DataFrame([varlist], columns = dim_df.columns, index = ['var'])
mct = dim_df.append(df)
mct

carat  \
0                                                   0.23   
1                                                   0.21   
2                                                   0.23   
3                                                   0.29   
4                                                   0.31   
...                                                  ...   
53936                                               0.72   
53937                                               0.70   
53938                                               0.86   
53939                                               0.75   
var    0       0.23
1       0.21
2       0.23
3      ...   

                                                   depth  \
0                                                  61.50   
1                                                  59.80   
2                                                  56.90   
3                                                  62.40   
4                                                  63.30   
...                                                  ...   
53936                                              63.10   
53937                                              62.80   
53938                                              61.00   
53939                                              62.20   
var    0       61.50
1       59.80
2       56.90
3   ...   

                                                   table  \
0                                                  55.00   
1                                                  61.00   
2                                                  65.00   
3                                                  58.00   
4                                                  58.00   
...                                                  ...   
53936                                              55.00   
53937                                              60.00   
53938                                              58.00   
53939                                              55.00   
var    0       55.00
1       61.00
2       65.00
3   ...   

                                                   price  \
0                                                    326   
1                                                    326   
2                                                    327   
3                                                    334   
4                                                    335   
...                                                  ...   
53936                                               2757   
53937                                               2757   
53938                                               2757   
53939                                               2757   
var    0         326
1         326
2         327
3   ...   

                                                       x  \
0                                                   3.95   
1                                                   3.89   
2                                                   4.05   
3                                                   4.20   
4                                                   4.34   
...                                                  ...   
53936                                               5.69   
53937                                               5.66   
53938                                               6.15   
53939                                               5.83   
var    0       3.95
1       3.89
2       4.05
3      ...   

                                                       y  \
0                                                   3.98   
1                                                   3.84   
2                                                   4.07   
3                                                   4.23   
4                                                   4.35   
...                                                  ...   
53936                     

In [70]:
var_df

carat            0.22
depth            2.05
table            4.99
price   15,915,629.42
x                1.26
y                1.30
z                0.50
dtype: float64